crates/noirc_evaluator/src/ssa/acir_gen/acir_mem.rs

```diff
@@ -5,26 +5,31 @@ use crate::ssa::{
 };
 use acvm::acir::native_types::Witness;
 use iter_extended::vecmap;
-use std::collections::HashMap;
+use std::collections::BTreeMap;
 
-// maps memory address to expression
 #[derive(Default)]
-pub struct MemoryMap {
-    inner: HashMap<u32, InternalVar>,
+pub struct AcirMem {
+    // maps memory address to InternalVar
+    memory_map: BTreeMap<ArrayId, BTreeMap<u32, InternalVar>>,
 }
 
-impl MemoryMap {
+impl AcirMem {
+    // returns the memory_map for the array
+    pub fn array_map_mut(&mut self, array_id: ArrayId) -> &mut BTreeMap<u32, InternalVar> {
```

Comment:
Can we create a type alias with doc comments:

``` 
pub type MemoryMap = BTreeMap<u32, InternalVar>;
```

Reply:
Done, I have added a type that includes the map, the map is now only used explicitly in one place only.